bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
> But to prevent ignoreEventList to grow unstopped and leak, I now use Set instead of List. 

There are still cases where the list could grow unbounded when using a Set, since item states could come in infinite variations.  Switching to a Set reverts [this commit](https://github.com/openhab/openhab/pull/2298/commits/04b092096b2ec9fc45ce8c249d939fcb1e83da46).  I hope to have more time to contribute more thoughts on this (but not at the moment unfortunately).